File: 1044-find-common-characters/find-common-characters.java
class Solution {
    public List<String> commonChars(String[] words) {
        List<String>ans=new ArrayList<>();
        int[] minfreq=new int[26];
        Arrays.fill(minfreq,Integer.MAX_VALUE);
        for(String word:words){
             int freq[]=new int[26];
             for(char c:word.toCharArray()){
                freq[c-'a']++;
             }
             for(int i=0;i<26;i++){
                minfreq[i]=Math.min(minfreq[i],freq[i]);
             }

        }
        for(int i=0;i<26;i++){
            while(minfreq[i]>0){
                ans.add(String.valueOf((char)('a'+i)));
                minfreq[i]--;
            }
        }
        return ans;
      
        
    }
}

File: 1044-find-common-characters/find-common-characters.py
class Solution:
    def commonChars(self, words: List[str]) -> List[str]:
        minfreq=[float('inf')]*26
        for word in words:
            freq=[0]*26

            for ch in word:
                freq[ord(ch) - ord('a')] += 1
            for i in range(26):
                minfreq[i]=min(minfreq[i],freq[i])
        
        ans=[]

        for i in range(26):
            while minfreq[i]>0:
                ans.append(chr(i+ord('a')))
                minfreq[i]-=1
        return ans